pkg/backend/display/progress.go

```diff
@@ -274,6 +274,63 @@ func ShowProgressEvents(op string, action apitype.UpdateKind, stack tokens.Stack
 	close(done)
 }
 
+// RenderProgressEvents renders the engine events as if to a terminal, providing a simple interface
+// for rendering the progress of an update.
+//
+// A "simple" terminal is used which does not render control sequences. The simple terminal's output
+// is written to opts.Stdout.
+//
+// For consistent output, these settings are enforced:
+//
+//	opts.Color = colors.Never
+//	opts.RenderOnDirty = false
+//	opts.IsInteractive = true
+func RenderProgressEvents(
+	op string,
+	action apitype.UpdateKind,
+	stack tokens.StackName,
+	proj tokens.PackageName,
+	permalink string,
+	events <-chan engine.Event,
+	done chan<- bool,
+	opts Options,
+	isPreview bool,
+	width, height int,
+) {
```

Comment:
I'm kind of wondering if we should go even further.

Should we remove the `opts` parameter entirely and just have this method return a string?

Nice to be able to take a `writer`--that leaves the possibility open for streaming--but do we even need any of the other options?

I'm slightly uncomfortable with the name, but I don't have a better suggestion. I think that the docs are likely sufficient.

Reply:
Inclined to agree with removing the `opts` if we're not respecting most of them.